test/logflare_web/controllers/endpoints_controller_test.exs

```diff
@@ -109,6 +109,52 @@ defmodule LogflareWeb.EndpointsControllerTest do
     end
   end
 
+  describe "bigquery with labels" do
+    setup do
+      _plan = insert(:plan, name: "Free")
+      user = insert(:user)
+      source = build(:source, user: user)
+      {:ok, user: user, source: source}
+    end
+
+    test "reference params in label, my_label=@my_param", %{
+      conn: conn,
+      user: user,
+      source: source
+    } do
+      pid = self()
+
+      GoogleApi.BigQuery.V2.Api.Jobs
+      |> stub(:bigquery_jobs_query, fn _conn, _proj_id, opts ->
+        send(pid, opts[:body].labels)
+        {:ok, TestUtils.gen_bq_response()}
+      end)
+
+      endpoint =
+        insert(:endpoint,
+          user: user,
+          enable_auth: true,
+          query: "with a as (select 1 as b) select b from a",
+          labels: "my_label=@my_param,other_value"
```

Comment:
besides `field=@param` I think we also allow `field=value`, maybe worth testing it here too just in case?